src/loaders/loaderUtils.test.ts

```diff
@@ -27,246 +31,319 @@ export const topicsResponseData: TopicData[] = [
   createTestTopicData(TEST_LOCAL_KAFKA_CLUSTER.id, "topic4", ["READ", "WRITE"]),
 ];
 
-describe("loaderUtils correlateTopicsWithSchemaSubjects() test", () => {
-  it("should correlate topics with schema subjects as strings", () => {
-    // topic 1-3 will be correlated with schema subjects, topic 4 will not.
-    // (Include empty string subject to further exercise issue #2149.)
-    const subjectStrings: string[] = ["topic1-value", "topic2-key", "topic3-Foo", ""];
-    const subjects: Subject[] = subjectStrings.map(
-      (name) =>
-        new Subject(
-          name,
-          TEST_LOCAL_SCHEMA_REGISTRY.connectionId,
-          TEST_LOCAL_SCHEMA_REGISTRY.environmentId,
-          TEST_LOCAL_SCHEMA_REGISTRY.id,
-        ),
-    );
-
-    const results = loaderUtils.correlateTopicsWithSchemaSubjects(
-      TEST_LOCAL_KAFKA_CLUSTER,
-      topicsResponseData,
-      subjects,
-    );
-
-    assert.ok(results[0].hasSchema);
-    assert.ok(results[1].hasSchema);
-    assert.ok(results[2].hasSchema);
-    assert.ok(!results[3].hasSchema);
-  });
-});
-
-describe("loaderUtils fetchSubjects() and fetchSchemasForSubject() tests", () => {
-  // Common suite and setup for loaderUtils functions that interact with SubjectsV1Api.
-
+describe("loaderUtils.ts", () => {
   let sandbox: sinon.SinonSandbox;
-  let stubbedSubjectsV1Api: sinon.SinonStubbedInstance<SubjectsV1Api>;
 
   beforeEach(() => {
     sandbox = sinon.createSandbox();
-
-    stubbedSubjectsV1Api = sandbox.createStubInstance(SubjectsV1Api);
-
-    const getSidecarStub: sinon.SinonStub = sandbox.stub(sidecar, "getSidecar");
-
-    const mockHandle = {
-      getSubjectsV1Api: () => {
-        return stubbedSubjectsV1Api;
-      },
-    };
-    getSidecarStub.resolves(mockHandle);
   });
 
   afterEach(() => {
     sandbox.restore();
   });
 
-  it("fetchSubjects() should return subjects sorted", async () => {
-    const subjectsRaw = ["Subject2", "subject3", "subject1"];
-    stubbedSubjectsV1Api.list.resolves(subjectsRaw);
+  describe("correlateTopicsWithSchemaSubjects()", () => {
+    it("should correlate topics with schema subjects as strings", () => {
+      // topic 1-3 will be correlated with schema subjects, topic 4 will not.
+      // (Include empty string subject to further exercise issue #2149.)
```

Comment:
related change? seems closed https://github.com/confluentinc/vscode/issues/2149

Reply:
Alas, the diff reads real ugly here. Converted this test suite to have a common single describe() with a single sandbox, then renamed some tests to be less redundant. No tests were removed, "should correlate topics with schema subjects as strings" is still the first test.

New tests for the cache key generation code added at the bottom.